homeassistant/components/trafikverket_ferry/coordinator.py

```diff
@@ -0,0 +1,93 @@
+"""DataUpdateCoordinator for the Trafikverket Weather integration."""
+from __future__ import annotations
+
+from datetime import date, datetime, time, timedelta
+import logging
+from typing import Any
+
+from pytrafikverket import TrafikverketFerry
+from pytrafikverket.trafikverket_ferry import FerryStop
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY, CONF_WEEKDAY, WEEKDAYS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+from homeassistant.util.dt import UTC, as_utc, parse_time
+
+from .const import CONF_FROM, CONF_TIME, CONF_TO, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+TIME_BETWEEN_UPDATES = timedelta(minutes=5)
+
+
+def next_weekday(fromdate: date, weekday: int) -> date:
```

Comment:
Can this be moved to the library?

Reply:
Eventually yes. I'm in the process to "get my hands on it" as it's also related to the other "trafikverket" integrations.
It need an update for sure

Reply:
I suggest we take this in follow-up